Redlib install error

### ✅ Have you read and understood the above guidelines?

Yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

Redlib

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/alpine-redlib.sh)"

### ⚙️ What settings are you using?

- [ ] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

_No response_

### 📈 Which Proxmox version are you on?

pve-manager/9.1.6/71482d1833ded40a (running kernel: 6.17.13-1-pve) root@pve:~# 

### 📝 Provide a clear and concise description of the issue.

During the install in verbose mode, the following error appears following the completion of Update Container OS step


### 🔄 Steps to reproduce the issue.

Execute te script using the provided shell command:
bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/alpine-redlib.sh)"


### ❌ Paste the full error output (if available).

OK: 36.4 MiB in 73 packages
  ✔️   Updated Container OS
  ✖️   in line 980: exit code 980 (Unknown error): while executing command sort -u
  ✖️   in line 980: exit code 980 (Unknown error): while executing command installed_pkgs=$(dpkg-query -W -f='${Package}
' 2> /dev/null | sort -u)

  --- Last 20 lines of log ---
[2026-03-09 00:43:26] [INFO] Setting up Container OS
[2026-03-09 00:43:26] [OK] Set up Container OS
[2026-03-09 00:43:26] [OK] Network Connected: 192.168.50.122
[2026-03-09 00:43:26] [OK] Internet: ✔ IPv4  DNS: 140.82.116.4
[2026-03-09 00:43:26] [INFO] Updating Container OS
[2026-03-09 00:43:28] [OK] Updated Container OS
[2026-03-09 00:43:28] [ERROR] in line 980: exit code 980 (Unknown error): while executing command sort -u
[2026-03-09 00:43:28] [ERROR] in line 980: exit code 980 (Unknown error): while executing command installed_pkgs=$(dpkg-query -W -f='${Package}
' 2> /dev/null | sort -u)
  -----------------------------------

  ✖️   Installation failed in container 122 (exit code: 980)
 📋 Installation log: /tmp/alpine-redlib-122-70d9c93a.log

  Error: Unknown error

### 🖼️ Additional context (optional).

_No response_

--- a/misc/tools.func
+++ b/misc/tools.func
@@ -969,13 +969,43 @@ verify_repo_available() {
 }
 
 # ------------------------------------------------------------------------------
-# Ensure dependencies are installed (with apt update caching)
+# Ensure dependencies are installed (with apt/apk update caching)
+# Supports both Debian (apt/dpkg) and Alpine (apk) systems
 # ------------------------------------------------------------------------------
 ensure_dependencies() {
   local deps=("$@")
   local missing=()
 
-  # Fast batch check using dpkg-query (much faster than individual checks)
+  # Detect Alpine Linux
+  if [[ -f /etc/alpine-release ]]; then
+    for dep in "${deps[@]}"; do
+      if command -v "$dep" &>/dev/null; then
+        continue
+      fi
+      if apk info -e "$dep" &>/dev/null; then
+        continue
+      fi
+      missing+=("$dep")
+    done
+
+    if [[ ${#missing[@]} -gt 0 ]]; then
+      $STD apk add --no-cache "${missing[@]}" || {
+        local failed=()
+        for pkg in "${missing[@]}"; do
+          if ! $STD apk add --no-cache "$pkg" 2>/dev/null; then
+            failed+=("$pkg")
+          fi
+        done
+        if [[ ${#failed[@]} -gt 0 ]]; then
+          msg_error "Failed to install dependencies: ${failed[*]}"
+          return 1
+        fi
+      }
+    fi
+    return 0
+  fi
+
+  # Debian/Ubuntu: Fast batch check using dpkg-query
   local installed_pkgs
   installed_pkgs=$(dpkg-query -W -f='${Package}\n' 2>/dev/null | sort -u)
 
@@ -1072,11 +1102,15 @@ create_temp_dir() {
 }
 
 # ------------------------------------------------------------------------------
-# Check if package is installed (faster than dpkg -l | grep)
+# Check if package is installed (supports both Debian and Alpine)
 # ------------------------------------------------------------------------------
 is_package_installed() {
   local package="$1"
-  dpkg-query -W -f='${Status}' "$package" 2>/dev/null | grep -q "^install ok installed$"
+  if [[ -f /etc/alpine-release ]]; then
+    apk info -e "$package" &>/dev/null
+  else
+    dpkg-query -W -f='${Status}' "$package" 2>/dev/null | grep -q "^install ok installed$"
+  fi
 }
 
 # ------------------------------------------------------------------------------

